docs/experimental/bicep-mcp-server.md

```diff
@@ -0,0 +1,46 @@
+# Using Bicep MCP Server in VS Code (Preview!)
+
+> [!NOTE]
+> This feature is currently in Preview while we collect feedback.
+
+## What is it?
+The Bicep VS Code extension now includes a Bicep MCP server with agentic tools to support Bicep code generation for AI agents in VS Code. Model Context Protocol (MCP) is an open standard that enables AI models to interact with external tools and services through a unified interface. 
+
+### What is MCP? 
+MCP follows a client-server architecture:
+- MCP clients (like VS Code) connect to MCP servers and request actions on behalf of the AI model.
+- MCP servers provide one or more tools that expose specific functionalities through a well-defined interface.
+- The Model Context Protocol (MCP) defines the message format for communication between clients and servers, including tool discovery, invocation, and response handling
+
+### Available Bicep MCP tools 
+- `ListAzResourceTypesForProvider`: Lists all available Azure resource types for a specific provider.The return value is a newline-separated list of resource types including their API version, e.g. `Microsoft.KeyVault/vaults@2024-11-01`.Such information is the most accurate and up-to-date as it is sourced from the Azure Resource Provider APIs. 
```

Comment:
The MCP SDK actually changes the names from camel case to snake case. Here's where you can see what the actual tool names will be:
https://github.com/Azure/bicep/blob/eec9d02126374ced168053508a9e81dabbdab828/src/Bicep.McpServer.UnitTests/ServerTests.cs#L26-L28

* `get_az_resource_type_schema`
* `get_bicep_best_practices`
* `list_az_resource_types_for_provider`